### PR TITLE
Important fixes for rubocop and stickler troubleshooting

### PR DESCRIPTION
**Pull request by nidalaa**

### CODE REVIEW

**Comment by nidalaa** [comment]
Hi @Nazeh!

As you reported problems here https://github.com/microverseinc/linters-config/pull/8 - could you test rubocop.yml from this branch in your projects?

**Comment by Nazeh** [comment]
Hi @nidalaa 

I tested the .rubocop.yml on two rails repositories and they seem to work fine.

One small issue In the ruby/README.md :
>3. In first commit of your feature branch add a copy of [.stickler.yml](./.stickler.yml).
    - **Remember** to use the copy linked above
    - **Remember** that file names starts with a dot
>3. In first commit of your feature branch add a copy of [.stickler.yml](./.stickler.yml) and [.rubocop.yml](./.rubocop.yml) .
    - **Remember** to use both files linked above
    - **Remember** that `.stickler.yml` and `.rubocop.yml` file names starts with a dot

I think number 3 should be deleted? and "file names **start** with a dot"

Thanks for your quick reply, I will use this in my current project.

File: ruby/README.md [hunk 1]
```diff
@@ -9,10 +9,18 @@ Please do the following **steps in this order**:
 3. In first commit of your feature branch add a copy of [.stickler.yml](./.stickler.yml).
     - **Remember** to use the copy linked above
     - **Remember** that file names starts with a dot
```
Comment (Nazeh) [inline ruby/README.md]: Line 9 - 11 are supposed to be deleted?

File: ruby/README.md [hunk 2]
```diff
@@ -9,10 +9,18 @@ Please do the following **steps in this order**:
 3. In first commit of your feature branch add a copy of [.stickler.yml](./.stickler.yml).
```
Comment (Nazeh) [inline ruby/README.md]:
```suggestion

```

File: ruby/README.md [hunk 3]
```diff
@@ -9,10 +9,18 @@ Please do the following **steps in this order**:
...
     - **Remember** to use the copy linked above
     - **Remember** that file names starts with a dot
+3. In first commit of your feature branch add a copy of [.stickler.yml](./.stickler.yml) and [.rubocop.yml](./.rubocop.yml) .
+    - **Remember** to use both files linked above
+    - **Remember** that `.stickler.yml` and `.rubocop.yml` file names starts with a dot
```
Comment (Nazeh) [inline ruby/README.md]:
```suggestion
    - **Remember** that `.stickler.yml` and `.rubocop.yml` file names start with a dot
```

File: ruby/.rubocop.yml [hunk 4]
```diff
@@ -1,12 +1,43 @@
...
-# Too short methods lead to extraction of single-use methods, which can make
-# the code easier to read (by naming things), but can also clutter the class
 Metrics/MethodLength:
   Include: ["app/controllers/*"]
   Max: 20
-
+Metrics/AbcSize:
+  Exclude: ["app/controllers/*"]
+  Max: 20
+Metrics/AbcSize:
```
Comment (Nazeh) [inline ruby/.rubocop.yml]: The duplication of **Metrics/AbcSize:**  gives an _error: duplicate key_ 
also, when running
```console
rubocop
``` 
It mentions:

```console
.rubocop.yml:15: `Metrics/AbcSize` is concealed by line 18
```

File: ruby/.rubocop.yml [hunk 5]
```diff
@@ -1,12 +1,43 @@
-# Commonly used screens these days easily fit more than 80 characters.
+AllCops:
+  DisplayCopNames: true
+  Exclude:
+    - "bin/*"
+    - "config/**/*"
```
Comment (Nazeh) [inline ruby/.rubocop.yml]: **Update**

After actually pushing changes to Github, some failures appeared that doesn't appear in the local environments. 

This happened to both repositories.

They are all problems with the **following Cop**:

```yml
C: Layout/ExtraSpacing: Unnecessary spacing detected.
```

And all happen in:

```yml
config/****.rb
```

and

```yml
config/environments/**.rb
```

**Quick fix:**
delete line 5, running automatic correction, then add it back or not.
It seems to block line 37 in the local environment for the config folder! but allow it for Stickler bot!

* Deleting rubocob.yml temporrely and performing auto-correction with no exceptions is not a bad Idea in general!

File: ruby/README.md (hunk 3 above)
Comment (nidalaa) [reply ruby/README.md]: <3 Thanks!

File: ruby/.rubocop.yml (hunk 5 above)
Comment (nidalaa) [reply ruby/.rubocop.yml]: So, I forked your micro-reddit repo and:

- bad news - I found out that Stickler ignores `exclude` in .rubocop.yml :)
- good news - there is an option to exclude files in .stickler.yml

I made sure they behave in the same way. Thanks for report!

File: ruby/.rubocop.yml (hunk 4 above)
Comment (nidalaa) [reply ruby/.rubocop.yml]: 👍

**Comment by nidalaa** [comment]
Hi @Nazeh! Thanks for your comments and QA 🙇‍♀

I split this PR into 2 different PRs - the README files changes are here https://github.com/microverseinc/linters-config/pull/10.

And rubocop config stayed in current PR.

**Comment by Nazeh** [comment]
Hi @nidalaa, I tested the latest changes on what I think would be a representative sample app, with generated controllers, models, and migrations. 

Everything seems to work just fine, but you can check all the test steps and the stickler checks in the **[commits history](https://github.com/Nazeh/test_app/commits/master)**

**Comment by nidalaa** [comment]
Great @Nazeh ! Thank you and high five!
horses

**Pull request merged**